### PR TITLE
Additional compiler option first commit for src and app directories

### PR DESCRIPTION
**Pull request by bfussell**
Will do test/ and app/ next and then turn on add compiler options for default build.

### CODE REVIEW

File: src/parson.c [hunk 1]
```diff
@@ -488,10 +490,16 @@ static JSON_Status json_object_dotremove_internal(JSON_Object *object, const cha
     JSON_Object *temp_object = NULL;
     char *dot_pos = NULL;
     int name_len = 0;
+    char *name2 = NULL;
 
     /* SAFEC modified to use _s */
     name_len = strnlen_s(name, STRING_NAME_MAX);
-    strstr_s((char *)name, name_len, ".", 1, &dot_pos);
+    name2 = calloc(STRING_NAME_MAX, sizeof(char));
```
Comment (abkarcher) [inline src/parson.c]: Unsure I follow, so just double checking:
- is name2 ever filled with data? seems to be an empty buffer for its lifespan
-Is it possible to lose a char of data if name_len = STRING_NAME_MAX?  (aka should calloc size be STRING_NAME_MAX + 1)

Comment (bfussell) [reply src/parson.c]: Good catch, didn't cut/paste the strncpy_s to each instance, will add that and re-test.

File: app/app_utils.c [hunk 2]
```diff
@@ -201,10 +203,12 @@ static ACVP_RESULT totp(char **token, int token_max) {
 
     otp = bin % DIGITS_POWER[ACVP_TOTP_LENGTH];
 
-    // generate format string like "%06d" to fix digits using 0
-    sprintf(format, "%c0%ldd", '%', (long int)ACVP_TOTP_LENGTH);
-
-    sprintf((char *)token_buff, format, otp);
+    /* If we ever change the size then it will fail hard here */
+    if (ACVP_TOTP_LENGTH != 8) {
```
Comment (abkarcher) [inline app/app_utils.c]: Is this something we may want to make a #define for?

Comment (bfussell) [reply app/app_utils.c]: Will address in next PR, thanks

**Pull request merged**